spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/ContainerInitializationTests.java

```diff
@@ -0,0 +1,128 @@
+
+import static org.hamcrest.Matchers.instanceOf;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.amqp.rabbit.listener.adapter.MessageListenerAdapter;
+import org.springframework.amqp.rabbit.listener.exception.FatalListenerStartupException;
+import org.springframework.amqp.rabbit.test.BrokerRunning;
+import org.springframework.context.ApplicationContextException;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * @author Gary Russell
+ * @since 1.6
+ *
+ */
+public class ContainerInitializationTests {
+
+	@Rule
+	public BrokerRunning brokerRunning = BrokerRunning.isRunningWithEmptyQueues("test.mismatch");
+
+	@After
+	public void tearDown() {
+		brokerRunning.removeTestQueues();
+	}
+
+	@Test
+	public void testMismatchedQueue() {
+		try {
+			AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(Config1.class);
+			context.close();
+			fail("expected initialization failure");
+		}
+		catch (ApplicationContextException e) {
+			assertThat(e.getCause(), instanceOf(FatalListenerStartupException.class));
+		}
+	}
+
+	@Test
+	public void testMismatchedQueueDuringRestart() throws Exception {
+		AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(Config2.class);
+		RabbitAdmin admin = context.getBean(RabbitAdmin.class);
+		admin.deleteQueue("test.mismatch");
+		admin.declareQueue(new Queue("test.mismatch", false, false, true));
+		SimpleMessageListenerContainer container = context.getBean(SimpleMessageListenerContainer.class);
+		int n = 0;
+		while (n++ < 100 && container.isRunning()) {
+			Thread.sleep(100);
+		}
+		assertFalse(container.isRunning());
+		context.close();
+	}
+
+	@Configuration
+	public static class Config1 {
+
+		@Bean
+		public ConnectionFactory connectionFactory() {
+			return new CachingConnectionFactory("localhost");
+		}
+
+		@Bean
+		public RabbitAdmin admin() {
+			return new RabbitAdmin(connectionFactory());
+		}
+
+		@Bean
+		public Queue queue() {
+			return new Queue("test.mismatch", false, false, true); // mismatched
+		}
+
+		@Bean
+		public SimpleMessageListenerContainer container() {
+			SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactory());
+			container.setQueueNames("test.mismatch");
+			container.setMessageListener(new MessageListenerAdapter(new Object() {
```

Comment:
What is the purpose of this extra `new Object()`?
Why regular `new MessageListener() {}` isn't enough ?

Minor, of course, but curious.